trail/upload/serializers.py

```diff
@@ -0,0 +1,34 @@
+from .models import Metadata
+from rest_framework import serializers
+
+
+class DynamicMetadataSerializer(serializers.ModelSerializer):
+    """
+    A ModelSerializer that takes an additional `fields` argument that
+    controls which fields should be displayed.
+    """
```

Comment:
This could use an Arguments section because otherwise nobody knows what key we need to specify the return fields under and the general signature of init means nothing will be documented.

```
Arguments
---------------
fields : `list[str]`
    List of keys that will be serialized into a response.